### PR TITLE
Separating local server code from test files and use it as a module.

### PR DESCRIPTION
**Pull request by vikiCoder**

### CODE REVIEW

File: test/integration/protocol-profile-behavior/disableBodyPruning.test.js [hunk 1]
```diff
@@ -64,7 +41,7 @@ describe('protocolProfileBehavior', function () {
             });
 
             after(function () {
-                testrun = rawRequest = null;
+                testrun = null;
```
Comment (codenirvana) [inline test/integration/protocol-profile-behavior/disableBodyPruning.test.js]: This is probably not required, remove `after` block as well.

File: test/integration/protocol-profile-behavior/disableBodyPruning.test.js [hunk 2]
```diff
@@ -132,7 +109,8 @@ describe('protocolProfileBehavior', function () {
                 sinon.assert.calledOnce(testrun.response);
                 sinon.assert.calledWith(testrun.response.getCall(0), null);
 
-                var response = rawRequest; // raw request message for this request
+                // raw request message for this request
+                var response = testrun.request.getCall(0).args[2].headers.get('raw-request');
```
Comment (codenirvana) [inline test/integration/protocol-profile-behavior/disableBodyPruning.test.js]: split this into response and rawResponse assignment to make it easier to understand.

File: test/integration/protocol-profile-behavior/disableBodyPruning.test.js [hunk 3]
```diff
@@ -1,41 +1,18 @@
...
 describe('protocolProfileBehavior', function () {
-    var server,
-        testrun,
-        rawRequest,
+    var testrun,
         PORT = 5050,
         URL = 'http://localhost:' + PORT;
 
     before(function (done) {
-        // Echo raw request message to handle body for http methods (GET, HEAD)
```
Comment (codenirvana) [inline test/integration/protocol-profile-behavior/disableBodyPruning.test.js]: Keep the comments added for context.

File: test/integration/sanity/http-methods.test.js [hunk 4]
```diff
@@ -1,43 +1,19 @@
...
 
 describe('http methods', function () {
-    var server,
-        testrun,
+    var testrun,
         PORT = 5050,
         URL = 'http://localhost:' + PORT;
 
     before(function (done) {
-        // Echo raw request message to handle custom http methods
```
Comment (codenirvana) [inline test/integration/sanity/http-methods.test.js]: Keep the comments here as well.

File: test/integration/protocol-profile-behavior/disableBodyPruning.test.js (hunk 2 above)
Comment (vikiCoder) [reply test/integration/protocol-profile-behavior/disableBodyPruning.test.js]: Even split lines without headers like below?
`var response = testrun.request.getCall(0).args[2].stream.toString();`

File: test/integration/protocol-profile-behavior/disableBodyPruning.test.js [hunk 5]
```diff
@@ -1,41 +1,18 @@
-var net = require('net'),
-    sinon = require('sinon'),
+var sinon = require('sinon'),
     expect = require('chai').expect,
-    enableServerDestroy = require('server-destroy');
+    echoServer = require('../../fixtures/server').createRawEchoServer();
```
[the remark this answers is not in the document]
Comment (vikiCoder) [reply test/integration/protocol-profile-behavior/disableBodyPruning.test.js]: Done.

**Pull request merged**